### ISSUE TITLE
Rsuite Next js 13 (appDir) is not compatible

### ISSUE THREAD

**Issue by vitalijalbu** [rsuite/rsuite#2937]
### What version of rsuite are you using?

latest

### What version of React are you using?

latest

### What version of TypeScript are you using (if any)?

no

### What browser are you using?

chrome

### Describe the Bug

I just installed a new next js and wanted to switch to their new appDir feature. Created all files but when want to import a simple button from rsuite it gives me error.

Taked inspiration from this repo: https://github.com/shadcn/taxonomy/blob/main/app/layout.tsx

My code layout.jsx:

![carbon](https://user-images.githubusercontent.com/40239902/203523900-22843e9a-e28f-47e2-b3ee-71904e3a6b3b.png)

My code page.jsx:

![carbon (1)](https://user-images.githubusercontent.com/40239902/203523912-a4981986-7f9a-4732-9f9e-c9b4822bde41.png)

Result:
![Screenshot 2022-11-23 alle 11 19 32](https://user-images.githubusercontent.com/40239902/203522735-f34557cf-163a-4f78-bc34-d57f82737e63.png)

### Expected Behavior

Normal expect working

### To Reproduce

1. create a next 13 app
2. set appDir true in next config
3. create layout & page files inside app folder
4. import rsuite button

**Comment by vitalijalbu** [comment]
working demo: https://codesandbox.io/s/rsuite-nextjs-hgpbkh
LOCALLY: doesn't work

**Comment by vitalijalbu** [comment]
Maybe is for node version? locally i have 18 but on codesandbox it runs on 14!?

**Comment by vitalijalbu** [comment]
@simonguo can you help please?? it's quite urgent...

**Comment by simonguo** [comment]
Hi @vitalijalbu 
Wait a minute, let's see if we can reproduce the error locally.

**Comment by simonguo** [comment]
The `appDir` function requires a minimum version of nodejs v16.8.0, you can modify the version by creating a `sandbox.config.json` [configuration file](https://codesandbox.io/docs/learn/sandboxes/configuration).

![image](https://user-images.githubusercontent.com/1203827/203745172-46d66374-45d1-4c73-a0f4-f347bf8c56e2.png)

**Comment by simonguo** [comment]
There is a solution to add `"use client"` to the head of the page.jsx file to let nextjs render on the client side.
```jsx
"use client";

import { Button } from "rsuite";

const Page = () => {
  return (
    <div className="show-fake-browser Page-page">
      ciao demo
      <Button appearance="primary">Hello World</Button>
    </div>
  );
};

export default Page;

```
![image](https://user-images.githubusercontent.com/1203827/203770649-de9da940-8a9b-4c95-9768-c2b05064caa5.png)

Complete example: https://codesandbox.io/s/rsuite-nextjs-forked-0pgwyu

**Comment by iska9der** [comment]
> I have been able to use rsuite in the latest version of next.js (13.4.15) using their app router. The problem is that I have to define "use client" at the top of every component that imports an rsuite component. This makes almost every component in the app a client component which is not ideal in next.js. Alternatively, I could wrap each rsuite component in another component that declares "use client", but that's a lot of work. It would be great if all rsuite components declared "use client" so I could freely import them in my server components. This is what MUI did in all of their components. This should have no affect on rsuite components in other environments and it should be a very safe change. Next.js is a very popular and powerful framework and I think making rsuite more compatible in that environment would be beneficial for the community. I think rsuite is the best UI library out there and I would love to use it in my next.js projects. Can the team update the library with "use client" in all components? Or if I open a pull request with this change, will it be accepted?

Maintainers, please, pay attention to this message

**Comment by SevenOutman** [comment]
> I have been able to use rsuite in the latest version of next.js (13.4.15) using their app router. The problem is that I have to define "use client" at the top of every component that imports an rsuite component. This makes almost every component in the app a client component which is not ideal in next.js. Alternatively, I could wrap each rsuite component in another component that declares "use client", but that's a lot of work. It would be great if all rsuite components declared "use client" so I could freely import them in my server components. This is what MUI did in all of their components. This should have no affect on rsuite components in other environments and it should be a very safe change. Next.js is a very popular and powerful framework and I think making rsuite more compatible in that environment would be beneficial for the community. I think rsuite is the best UI library out there and I would love to use it in my next.js projects. Can the team update the library with "use client" in all components? Or if I open a pull request with this change, will it be accepted?

@psebestyen Thanks for the detailed explanation! As we don't have adopted RSC in our apps, we didn't investigate how to make our library RSC-compatible. It's welcomed and appreciated if you could contribute! And it's certainly going to be accepted once we make sure it's a safe change.

**Comment by iska9der** [comment]
@psebestyen 
Everything below is my guesses:
is it not enough to insert at the top of `rsuite/src/index.tsx`  `"use client"` directive?

Pros: it doesn't take long, just adding one line in one file.

Cons: we can achieve what we want, importing like this: 
```
import { Button } from "rsuite";
```
but if we import like this:
```
import Button from "rsuite/esm/Button";
```
we will catch the same behavior as now

**Comment by psebestyen** [comment]
@iska9der, good point. It's a one liner and goes a long way. I will test it and make the change if all goes well.

**Comment by simonguo** [comment]
Try add rsuite to an external that requires a react/client context

```ts
/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    serverComponentsExternalPackages: ['rsuite'],
  },
}
 
module.exports = nextConfig
```

**Comment by nikiforov-org** [comment]
The same issue in the latest NEXT.JS 14.1.0, nothing works without 'use client' as well
